File: packages/kt-client/src/main/kotlin/com/google/firebase/auth/OAuthCredential.kt
package com.google.firebase.auth

class OAuthCredential internal constructor(
    override val provider: String,
    val idToken: String?,
    val accessToken: String?,
    val rawNonce: String? = null
) : AuthCredential(provider) {
    override fun toWireMap(): Map<String, Any?> = mapOf(
        "providerId" to provider,
        "idToken" to idToken,
        "accessToken" to accessToken,
        "rawNonce" to rawNonce
    )
}

File: packages/swift-client/Sources/FirebaseAuth/OAuthCredential.swift
import Foundation
import PyricFirestore

public final class OAuthCredential: AuthCredential, @unchecked Sendable {
    public let idToken: String?
    public let accessToken: String?
    public let rawNonce: String?

    public init(providerID: String, idToken: String?, rawNonce: String?, accessToken: String?) {
        self.idToken = idToken
        self.accessToken = accessToken
        self.rawNonce = rawNonce
        super.init(provider: providerID)
    }

    override internal func toWireParams() -> [String: AnySendable] {
        var params: [String: AnySendable] = ["providerId": .string(provider)]
        if let idToken { params["idToken"] = .string(idToken) }
        if let accessToken { params["accessToken"] = .string(accessToken) }
        if let rawNonce { params["rawNonce"] = .string(rawNonce) }
        return params
    }
}
